cishouseholds/derive.py

```diff
@@ -810,14 +814,22 @@ def assign_date_difference(
         First date column name.
     end_reference_column
         Second date column name.
-
+    format
+        time format (days, weeks, months)
```

Comment:
This suggests that the function supports `"months"` but there doesn't seem to be any functionality for that below. Please also add validation that the user has supplied a supported format, as the code below will treat `format="potato"` as `"days"`

Reply:
added validation

Reply:
> Please could you give this a run in the dev environment, to make sure the union_dependent_transformation runs ok?
> 
> Other than these two comments, looks good to me

we will create a new ticket for this as it is part of a larger bug